sktime/datatypes/_panel/_check.py

```diff
@@ -171,41 +173,56 @@ def check_pdmultiindex_panel(obj, return_metadata=False, var_name="obj"):
         msg = f"{var_name} must have a MultiIndex with 2 levels, found {nlevels}"
         return _ret(False, msg, None, return_metadata)
 
+    # check that no dtype is object
+    if "object" in obj.dtypes.values:
+        msg = f"{var_name} should not have column of 'object' dtype"
+        return _ret(False, msg, None, return_metadata)
+
+    # check whether the time index is of valid type
+    if not is_in_valid_index_types(obj.index.get_level_values(-1)):
+        msg = (
+            f"{type(obj.index)} is not supported for {var_name}, use "
+            f"one of {VALID_INDEX_TYPES} or integer index instead."
+        )
+        return _ret(False, msg, None, return_metadata)
+
+    # inst_inds_names = obj.index.names[0:-1]
+    # time_inds_names = obj.index.names[-1]
+    time_obj = obj.reset_index(-1).drop(obj.columns, axis=1)
+    time_grp = time_obj.groupby(level=0, group_keys=True, as_index=True)
+    inst_inds = time_obj.index.unique()
+
     # check instance index being integer or range index
-    instind = obj.index.get_level_values(0)
-    if not is_in_valid_multiindex_types(instind):
+    if not is_in_valid_multiindex_types(inst_inds):
         msg = (
             f"instance index (first/highest index) must be {VALID_MULTIINDEX_TYPES}, "
-            f"integer index, but found {type(instind)}"
+            f"integer index, but found {type(inst_inds)}"
         )
         return _ret(False, msg, None, return_metadata)
 
-    inst_inds = obj.index.get_level_values(0).unique()
-    # inst_inds = np.unique(obj.index.get_level_values(0))
+    is_equally_spaced = (
+        time_grp.diff().groupby(level=0, group_keys=True, as_index=True).nunique()
+    )
 
-    check_res = [
-        check_pddataframe_series(obj.loc[i], return_metadata=True) for i in inst_inds
-    ]
-    bad_inds = [i for i in range(len(inst_inds)) if not check_res[i][0]]
+    unique_diff = is_equally_spaced.iloc[:, 0].unique()
+    is_equal_length = time_grp.count()
```

Comment:
Nit, feel free to ignore this: I found the naming of `is_equal_length` and `is_equally_spaced` difficult to follow. They both represent the result of applying an aggregation function and return a single value for each instance. `is_equal_length 
 = time_grp.count()` returns the length of each instance and ` is_equally_spaced` represents the number of unique time deltas (ignoring NaNs) found for each instance. Potentially 'len_by_instance' and 'nunique_time_deltas_by_instance' might be more explicit, however, far more verbose and perhaps not as clear to everyone else as it is to me.

Reply:
too late... these are used throughout the code so it would be risky.
Might be worth renaming once we refactor the checks though.
To avoid forgetting it, a good place to collect ideas is here: https://github.com/sktime/sktime/issues/3512

Reply:
I just mean the use of `is_equal_length` and `is_equally_spaced` as intermediate variables at these particular lines of code. I could follow them fine elsewhere in this module (e.g., as name of the key in the `metadata` dictionary). Unless you're suggesting they're also used as variable names in other modules.

Reply:
I agree that this wording would be a lot better, we should keep that for a different PR since it would cover more checks and we need to be consistent.